### PR TITLE
Fix type issues in `Hero`

### PR DESCRIPTION
**Pull request by TylerJDev**
## Summary

Adjusts types for `HeroActions` type. 
* Remove `Omit` for `variant` prop
* Fix `leadingVisual` and `trailingVisual` type issue

## List of notable changes:

- Adjusts TS types for `HeroActions`

## What should reviewers focus on?

- The ability to use props stemming from `Button`
- Added story to showcase icon support and `variant` usage

## Steps to test:

* Ensure no TypeScript errors are present when utilizing props from the [Brand `Button` component](https://primer-5dccbf8f09-26139705.drafts.github.io/components/Button#component-props).

## Contributor checklist:

- [x] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### CODE REVIEW

File: packages/react/src/Hero/Hero.tsx [hunk 1]
```diff
@@ -74,7 +74,9 @@ function HeroDescription({size = '400', weight, children}: PropsWithChildren<Her
 
 type HeroActions = {
   href: string
-} & Omit<ButtonBaseProps, 'variant'>
+  leadingVisual?: React.ReactNode
```
Comment (rezrah) [inline packages/react/src/Hero/Hero.tsx:77]: Can these be picked from ButtonBaseProps instead? And any ideas why this isn't getting inherited? Feels redundant to maintain these props here too.

File: packages/react/src/Hero/Hero.tsx [hunk 2]
```diff
@@ -74,7 +74,9 @@ function HeroDescription({size = '400', weight, children}: PropsWithChildren<Her
...
+  trailingVisual?: React.ReactNode
+} & Omit<ButtonBaseProps, 'trailingVisual' | 'leadingVisual'>
```
Comment (TylerJDev) [inline packages/react/src/Hero/Hero.tsx:79]: This was one of the few ways I was able to make `leadingVisual` and `trailingVisual` play nice with Octicons with its `Icon` type. It seems that there's a mismatch in props against `ReactElement` that TypeScript was complaining about, which is only a problem when using `Button` here, and not directly. 

A alternative would be replacing `ReactElement` type in `Button` for `leadingVisual` and `trailingVisual` with `React.ReactNode`, but I only wanted to adjust the `Hero` component, and not change anything within `Button`.

File: packages/react/src/Hero/Hero.tsx (hunk 1 above)
Comment (TylerJDev) [reply packages/react/src/Hero/Hero.tsx:77]: I left a comment on this https://github.com/primer/brand/pull/284#discussion_r1200626396 - I'm not entirely sure why it's throwing an error here. I get the following TS error: `Type 'FunctionComponent<IconProps>' is missing the following properties from type 'ReactElement<any, string | JSXElementConstructor<any>>': type, props, key`, which isn't present when utilizing `Button` and the "visual" props directly.

Utilizing `React.ReactNode` fixed it, as it seems to be tied to `ReactElement`. I kept the change in this component via `Omit` as to not touch `Button`. I can do some more digging in the meantime though.

Comment (TylerJDev) [reply packages/react/src/Hero/Hero.tsx:77]: @rezrah - Digging into this a bit deeper it seems like this is tied to the `ReactElement` type in `Button` when utilizing an Octicon. ReactElement does not appear to apply when the type is a functional component `React.FC`, which is what `Octicon` seems to return as its type.

When used directly, `Button` doesn't seem to check the types of the props that are passed to it, which I think is why it doesn't cause an issue when `Button` is used directly with `leadingVisual` and `trailingVisual`. 

Could we replace the `ReactElement` type in `Button.tsx` with `React.ReactNode` or add both, (e.g. 
 `leadingVisual?: ReactElement | React.ReactNode`)? It seems like we do this within `Label` and `TextInput` too.
 
 Let me know your thoughts!

File: packages/react/src/Button/Button.tsx [hunk 3]
```diff
@@ -16,11 +18,11 @@ export type ButtonBaseProps = {
   /**
    * The leading visual appears before the button content
    */
-  leadingVisual?: ReactElement
+  leadingVisual?: ReactElement | Icon
```
Comment (rezrah) [inline packages/react/src/Button/Button.tsx:21]: @TylerJDev - here's the change ☝️

File: packages/react/src/Hero/Hero.tsx (hunk 1 above)
Comment (rezrah) [reply packages/react/src/Hero/Hero.tsx]: Hey Tyler, you're right that it needs to be fixed in Button. Putting this here for posterity, as we spoke about this over slack. The type needs to extend Icon from Octicons, doing that should fix it. I've pushed the change in [this commit](https://github.com/primer/brand/pull/284/commits/2a9a062f645cf71c7d687d9d6aa9f7dfd992a9ba).

**Pull request merged**